homeassistant/components/plugwise_stick/config_flow.py

```diff
@@ -0,0 +1,131 @@
+"""Config flow for the Plugwise_stick platform."""
+import os
+from typing import Dict
+
+import plugwise
+from plugwise.exceptions import NetworkDown, PortError, StickInitError, TimeoutException
+import serial.tools.list_ports
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.core import HomeAssistant, callback
+
+from .const import CONF_USB_PATH, DOMAIN
+
+CONF_MANUAL_PATH = "Enter Manually"
+
+
+@callback
+def plugwise_stick_entries(hass: HomeAssistant):
+    """Return existing connections for Plugwise USB-stick domain."""
+    return {
+        (entry.data[CONF_USB_PATH])
+        for entry in hass.config_entries.async_entries(DOMAIN)
+    }
+
+
+class PlugwiseConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
+
+    async def async_step_user(self, user_input=None):
+        """Step when user initializes a integration."""
+        errors = {}
+        ports = await self.hass.async_add_executor_job(serial.tools.list_ports.comports)
+        list_of_ports = [
+            f"{p}, s/n: {p.serial_number or 'n/a'}"
+            + (f" - {p.manufacturer}" if p.manufacturer else "")
+            for p in ports
+        ]
+        list_of_ports.append(CONF_MANUAL_PATH)
+
+        if user_input is not None:
+            user_selection = user_input[CONF_USB_PATH]
+            if user_selection == CONF_MANUAL_PATH:
+                return await self.async_step_manual_path()
+            if user_selection in list_of_ports:
+                port = ports[list_of_ports.index(user_selection)]
+                device_path = await self.hass.async_add_executor_job(
+                    get_serial_by_id, port.device
+                )
+            else:
+                device_path = await self.hass.async_add_executor_job(
+                    get_serial_by_id, user_selection
+                )
```

Comment:
Suggestion: here you could add something like:
```
            for entry in self._async_current_entries():
                if entry.data.get("parameter") == user_input["parameter"]:
                    return self.async_abort(reason="already_configured")
```
This, together with the unique_id, see below, stops the possibility of adding a configuration 2 times.
This requires also a few addition lines in string.json.

Reply:
I do prevent adding the same USB-stick in a manually started second config flow at [L103-L105](https://github.com/home-assistant/core/pull/35713/files/c1f726f16a156dcedbc2df4228b5aa5b5b1e96ed#diff-b470aa8096bc3e4782af381d5b99f8d3R103-R105) together with [L18-L24](https://github.com/home-assistant/core/pull/35713/files/c1f726f16a156dcedbc2df4228b5aa5b5b1e96ed#diff-b470aa8096bc3e4782af381d5b99f8d3R18-R24) which gives a nice error message:
![image](https://user-images.githubusercontent.com/6314181/93260205-5836c080-f7a1-11ea-8dcb-78b6578deecc.png)

I'm not sure it is user friendly to abort the config flow instead. To me aborting the configflow seems to be only usefull when automatic discovery triggers the config flow, which is not the case for serial devices.

Reply:
Very good! Then please ignore my comment.